### ISSUE TITLE
importing models with mlflow-artifacts: source

### ISSUE THREAD

**Issue by djohnson-cpacket** [mlflow/mlflow-export-import#53]
Importing models that have a source that starts "mlflow-artifacts:" fails.  This is due to the fact that there is a file existence check override for "dbfs:" but not "mlflow-artifiacts:".

**Comment by djohnson-cpacket** [comment]
Ref https://github.com/mlflow/mlflow-export-import/pull/52

**Comment by amesar** [comment]
Good catch. I'm finishing up on another PR now. Will merge yours after that.

**Comment by djohnson-cpacket** [comment]
Thanks a lot for the feedback.

David.

[image: cPacket Networks | LinkedIn]

David Johnson

cPacket Networks

*Senior Systems Engineer*

M: 510.851.5846

***@***.*** | www.cpacket.com
<https://www.google.com/url?q=https://www.google.com/url?q%3Dhttp://www.cpacket.com/%26source%3Dgmail-imap%26ust%3D1621465754000000%26usg%3DAOvVaw1ZNt3rXwpPgk9kH1x3CotY&source=gmail-imap&ust=1622061570000000&usg=AOvVaw06MOY3uDpPiIi8HkoQNyq_>


*The content of this message is confidential. If you have received it by
mistake, please inform us by an email reply and then delete the message. It
is forbidden to copy, forward, or in any way reveal the contents of this
message to anyone. The integrity and security of this email cannot be
guaranteed over the Internet. Therefore, the sender will not be held liable
for any damage caused by the message.*

**Comment by amesar** [comment]
Tests passed and I approved the change.

**Comment by teoria** [comment]
Same problem with s3:// artifacts.
` if not dst_source.startswith(("dbfs:","s3:","mlflow-artifacts:")) and not os.path.exists(dst_source):
           raise MlflowExportImportException(f"'source' argument for MLflowClient.create_model_version does not exist: {dst_source}")
       `

**Comment by helpsystems-mushkevych** [comment]
Reporting same problem. 
@amesar  I noticed that #52 has been closed without a merge. 
Is there a chance for a quick resolution to this ticket?

**Comment by guizsantos** [comment]
Facing same issue here, with S3, thanks for the workaroung @teoria that got it working.

**Comment by ammar-nizami** [comment]
Facing the same issue with `s3`.

**Comment by dhirajdj30** [comment]
MlflowExportImportException(f"'source' argument for MLflowClient.create_model_version does not exist: {dst_source}", http_status_code=404)
